devices/HapticGlove/test/VisualizeFrames.cpp

```diff
@@ -0,0 +1,355 @@
+/*
+ * Copyright (C) 2021 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-2-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#include <iDynTree/Visualizer.h>
+#include <iDynTree/yarp/YARPConversions.h>
+
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/os/Network.h>
+#include <yarp/os/RFModule.h>
+#include <yarp/sig/Vector.h>
+
+#include <csignal>
+#include <cstdlib>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <vector>
+
+#include <Wearable/IWear/IWear.h>
+
+using namespace yarp::os;
+
+std::atomic<bool> isClosing{false};
+
+void my_handler(int) { isClosing = true; }
+
+#ifdef WIN32
+
+#include <windows.h>
+
+BOOL WINAPI CtrlHandler(DWORD fdwCtrlType) {
+  switch (fdwCtrlType) {
+    // Handle the CTRL-C signal.
+  case CTRL_C_EVENT:
+  case CTRL_CLOSE_EVENT:
+  case CTRL_SHUTDOWN_EVENT:
+    my_handler(0);
+    return TRUE;
+
+  // Handle all other events
+  default:
+    return FALSE;
+  }
+}
+#endif
+
+void handleSigInt() {
+#ifdef WIN32
+  SetConsoleCtrlHandler(CtrlHandler, TRUE);
+#else
+  struct sigaction action;
+  memset(&action, 0, sizeof(action));
+  action.sa_handler = &my_handler;
+  sigaction(SIGINT, &action, NULL);
+  sigaction(SIGTERM, &action, NULL);
+  sigaction(SIGABRT, &action, NULL);
+#endif
+}
+
+struct FrameViewier {
+  std::shared_ptr<FrameViewier> parent{nullptr};
+  std::string name;
+  size_t vizIndex;
+  iDynTree::Transform transform;
+};
+
+int main(int argc, char *argv[]) {
+  std::string portNameIn = "WearableData:I";
+  std::string portName = "WearableData";
+  std::string namePrefix = "/WearableViewer";
+  std::string logPrefix = "WearableDataVisualizer";
+
+  // prepare and configure the resource finder
+  yarp::os::ResourceFinder &config =
+      yarp::os::ResourceFinder::getResourceFinderSingleton();
+
+  config.setDefaultConfigFile("config.ini");
+
+  config.configure(argc, argv);
+
+  // =========================
+  // Start the visualizer
+  // =========================
+  iDynTree::Visualizer visualizer;
+
+  bool ok = visualizer.init();
+
+  visualizer.camera().animator()->enableMouseControl();
+
+  if (!ok) {
+    std::cerr << "Failed to initialize the visualizer." << std::endl;
+    return EXIT_FAILURE;
+  }
+  // =========================
+  // Open the Drivers
+  // =========================
+  yarp::dev::PolyDriver tfDriver, iWearDriver;
+  wearable::IWear *iWear;
+
+  yarp::os::Property wearOptions, tfClientCfg;
+
+  // Iwear Remapper
+  yarp::os::Value *wearableDataPort;
+  if (!config.check("wearable_data_ports", wearableDataPort)) {
```

Comment:
If the config parameter is a single port, better to have the associate variable also singular `wearable_data_port`

Reply:
NVM, I see the config file has wearable data ports from left and right. So, may be `wearableDataPorts`

Reply:
Consider that, here https://github.com/robotology/wearables/blob/c47205dae6c6c508f2f39545e2a7e835d3bf3610/devices/IWearRemapper/src/IWearRemapper.cpp#L129-L140 it accepts a list of strings not a string!

Reply:
@Yeshasvitvs  Can I resolve this conversation?